unison-cli/src/Unison/Codebase/Editor/HandleInput/FormatFile.hs

```diff
@@ -82,8 +82,9 @@ formatFile makePPEDForFile formattingWidth currentPath inputParsedFile inputType
             & over _2 Pretty.syntaxToColor
   formattedTerms <-
     (FileSummary.termsBySymbol fileSummary)
-      & Map.filter (\(_bindingAnn, _, trm, _) -> shouldFormatTerm (ABT.annotation trm) trm)
-      & itraverse \sym (_bindingAnn, mayRefId, trm, _typ) -> do
+      & fmap (\(bindingAnn, mayRefId, trm, _typ) -> (bindingAnn <> ABT.annotation trm, mayRefId, trm))
```

Comment:
Or, I guess you can recover it from the term later?

Reply:
Yeah the full span annotation can be recovered just like you see here, nothing is lost :) 

But in case you're wondering, I'm doing all sorts of very creative things to get as much annotation information as I can from a limited number of annotation slots; it's holding together so far, but ideally I'd add a lot more slots and label them better, but I can't do that inside terms without doing a huge migration :'(

Reply:
We can take a look at what it would take to do a migration, I don't think we should generally super scared of them; only scared on a case by case basis ideally :)

Reply:
Well, we've never done a codebase migration where Share is involved, so I'm a little scared at least 🙈